src/mongo/util/processinfo_win32.cpp

```diff
@@ -116,6 +116,75 @@ namespace mongo {
 #endif
     }
 
+    bool getFileVersion(const char *filePath, DWORD &fileVersionMS, DWORD &fileVersionLS) {
+        DWORD verSize = GetFileVersionInfoSizeA(filePath, NULL);
+        if (verSize == 0) {
+            DWORD gle = GetLastError();
+            warning() << "GetFileVersionInfoSizeA on " << filePath << " failed with " << errnoWithDescription(gle);
+            return false;
+        }
+
+        boost::scoped_array<char> verData(new char[verSize]);
+        if (GetFileVersionInfoA(filePath, NULL, verSize, verData.get()) == 0) {
+            DWORD gle = GetLastError();
+            warning() << "GetFileVersionInfoSizeA on " << filePath << " failed with " << errnoWithDescription(gle);
+            return false;
+        }
+
+        UINT size;
+        VS_FIXEDFILEINFO *verInfo;
+        if (VerQueryValueA(verData.get(), "\\", (LPVOID *)&verInfo, &size) == 0) {
+            DWORD gle = GetLastError();
+            warning() << "VerQueryValueA on " << filePath << " failed with " << errnoWithDescription(gle);
+            return false;
+        }
+    
+        if (size != sizeof(VS_FIXEDFILEINFO)) {
+            warning() << "VerQueryValueA on " << filePath << " returned structure with unexpected size";
+            return false;
+        }
+
+        fileVersionMS = verInfo->dwFileVersionMS;
+        fileVersionLS = verInfo->dwFileVersionLS;
+        return true;
+    }
+
+    // If the version of the ntfs.sys driver shows that the KB2731284 hotfix or a later update
+    // is installed, zeroing out data files is unnecessary. The file version numbers used below
+    // are taken from the Hotfix File Information at http://support.microsoft.com/kb/2731284.
+    bool isKB2731284OrLaterUpdateInstalled() {
+        char systemDirectory[MAX_PATH];
+        if (!GetSystemDirectoryA(systemDirectory, sizeof(systemDirectory))) {
```

Comment:
If you check the documentation for `GetSystemDirectory`, you'll see that the return value is not a Boolean but rather an integer.  If it fails due to the buffer not being large enough, it will return the required buffer size.  Therefore, a nonzero return value does not necessarily mean success.  I recognize it's unlikely that the system directory path will exceed `MAX_PATH` characters, but we should still check this return value properly.